token/program-2022/src/instruction.rs

```diff
@@ -85,11 +89,17 @@ pub enum TokenInstruction {
         /// account.
         m: u8,
     },
+    /// NOTE This instruction is deprecated in favor of `TransferChecked` or
+    /// `TransferCheckedWithFee`
+    ///
     /// Transfers tokens from one account to another either directly or via a
     /// delegate.  If this account is associated with the native mint then equal
     /// amounts of SOL and Tokens will be transferred to the destination
     /// account.
     ///
+    /// If the accounts contain `AccountTransferFeeExtension`s, this will fail.  Mints with
```

Comment:
```suggestion
    /// If either accounts contains an `AccountTransferFee` extension, this will fail.  Mints with
```

That is, can we just drop the `Extension` from the structs in general?  Contrast with `MintCloseAuthority